sec1_intro_and_setup/test_app/.browserslistrc

```diff
@@ -0,0 +1 @@
+defaults
```

Comment:
`defaults` means it will support the default browser sets by browserslist. The supported browsers are [here](https://github.com/browserslist/browserslist#browsers)

Question:
1. What does this mean?
![image](https://user-images.githubusercontent.com/60863955/126159102-65f61dfe-74bf-4215-bf1e-041a8c2af31e.png)
(Refer [this](https://github.com/browserslist/browserslist#browserslistrc) for more reference)

Reply:
1. Your link is not accurate. That shows the kind of browsers it supports. But what versions?
2. How do you check what versions is supported?
3. You said `it will support the default browser sets by browserslist`. What does `it` refers to?
4. Why do we need this?

---

To your question:
1. This means browserlist will check for the config file. In this case, `.browserslistrc` or other ways to write the config file in those places of your app

Reply:
@tlkiong 
To your question:

1. The `browserslist`  default browsers is (> 0.5%, last 2 versions, Firefox ESR, not dead) 
    - `> 0.5% `means all browsers wtih more than 0.5% of global market share.
    - `last 2 versions` means the last 2 versions for each browser.
    - `Firefox ESR` is the latest Firefox Extended Support Release browser version.
    - `not dead` means exclude browsers without official support or updates for 24 months.

2. Can check the browser version in `.browserslistrc` (Note that if one's hope to reduce config files in the project root, can specify browser versions in `package.json` file). Since there are no specified browsers in our case, the browser version will proceed with `defaults` which is stated in the `.browserslistrc` file.

3.  My original meaning that `it` refers to the `the environment chosen by the browserslist` but after I read more in details, seems like I get the wrong idea. To answer the question asked in the beginnning, `defaults` is the default browsers set by `.browserslistrc` file which is > 0.5%, last 2 versions, Firefox ESR, not dead.

4. Different browsers (and versions) support different web platform features, `browserslist` is a config that shares target browsers and Node.js versions between different front-end tools which allow us to make choices based on the support of those features balanced with the analytics about the browsers used by the users. 

Question:
1. According to number 4, is this why some of the features will no longer be supported when a machine did not upgrade its browsers?

Reply:
1. But not accurate. Go and find out how to get the exact **browser version** that is supported by the browserlist configuration. Your answer didn't tell me the versions
2. Completely wrong. I am looking for a command you can key in to check.
3. .
4. Good answer. 

---

To your question:
1. Why some feature no longer supported by a machine?
This statement is not accurate. There are some feature that is supported by some browsers, some not. How to check this? You can go to: https://caniuse.com/
Why is this so? cause each browser is a software belonging to someone. That someone decides to not want to support then it won't support. For example, look at [this](https://caniuse.com/mdn-css_properties_gap_grid_context). This is `property: gap` for CSS. IE don't want support. What can us as the devs do? We can choose to not support IE. Or if we **need** to support IE cause our main users are there, we would have to do some workarounds

Reply:
1.  
```
and_chr 91
and_ff 89  
and_qq 10.4
and_uc 12.12
android 91
baidu 7.12
chrome 91
chrome 90
edge 91
edge 90
firefox 90
firefox 89
firefox 78
ie 11
ios_saf 14.5-14.7
ios_saf 14.0-14.4
ios_saf 13.4-13.7
kaios 2.5
op_mini all
op_mob 62
opera 77
opera 76
safari 14.1
safari 14
samsung 14.0
samsung 13.0
```
2. `npx browserslist`